src/Workspaces/CoreTest/SolutionTests/SolutionTests.cs

```diff
@@ -690,8 +690,13 @@ private void StopObservingAndWaitForReferenceToGo(ObjectReference observed, int
             var actualTimePassed = DateTime.UtcNow - start + TimeSpan.FromMilliseconds(TimerPrecision);
             var isTargetCollected = observed.Weak.Target == null;
 
-            if (!isTargetCollected && !string.IsNullOrEmpty(dumpFileName))
+            if (!isTargetCollected)
             {
+                if (string.IsNullOrEmpty(dumpFileName))
```

Comment:
It can't be a required parameter because then every test that indirectly calls through this method would need to pipe through its name (or some other unique ID), but as I think about it more, maybe this shouldn't even be a parameter and an automatic name always gets created?  Thoughts?  Also @tannergooding because we were talking about this yesterday.